source/model.py

```diff
@@ -1270,11 +1270,13 @@ def __c__(self, writer):
 
 class Enumeration(Type):
 
-    def __init__(self, type_name, elems_list, enum_name = ""):
-        super(Enumeration, self).__init__(name = type_name, incomplete = False)
+    def __init__(self, elems_list, name = None):
+        super(Enumeration, self).__init__(name = name, incomplete = False)
+
+        if name is None:
+            self.name = "enum.auto" + str(id(self))
```

Comment:
Is `name` for `__str__` only?
If so, it's confusing. Better move the evaluation to `__str__`. In a `lazy` property, maybe.

Reply:
This `name` using by `Source.add_type` for registration. But I can’t always create a name, because the flag `in_named` is used to distinguish between named and unnamed enums.

Reply:
`Source.types` must use same keys as `Type.reg` do for same types. It looks like `Source` must support nameless types too...